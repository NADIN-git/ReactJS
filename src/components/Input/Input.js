import React from 'react';
import { TextField } from "@material-ui/core";
import { Button } from "@material-ui/core";
import { ThemeProvider, createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      main: "#6170fb",
    },
    secondary: {
      main: "#0098FF",
    },
  },
});

const Input = (props) => {
  const { onSubmit } = props
  const [inputValue, setInputValue] = React.useState('')
  
  const messageSubmit = (e) => {
    e.preventDefault()

    if (onSubmit) {
        onSubmit(inputValue)
        setInputValue('')
    }
}

  const messageUpdate = (e) => {
    setInputValue(e.target.value)
  }

  return (
    <ThemeProvider theme={theme}>
      <div>
        <form className="Message_border Left_right" onSubmit={messageSubmit}>
          <TextField
            color='primary'
            fullWidth
            required
            autoFocus
            placeholder="Введите сообщение"
            value={inputValue}
            onChange={messageUpdate}
            margin="normal"
            variant="filled"
          />
          <Button
            type="submit"
            style={{ margin: '20px' }}
            variant="contained"
            color="primary">
            Отправить
          </Button>
        </form>
      </div >
    </ThemeProvider>
  );
}

export default Input;
