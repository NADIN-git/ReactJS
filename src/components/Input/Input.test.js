import React from "react";
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer";
import Button from "./Input";
import TextField from "./Input";
import Input from "./Input";

it('Renders Input', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Input />, div)
  ReactDOM.unmountComponentAtNode(div)  
})

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});

describe("TextField component", () => {
  test("Matches the snapshot", () => {
    const textField = create(<TextField />);
    expect(textField.toJSON()).toMatchSnapshot();    
  });
});
