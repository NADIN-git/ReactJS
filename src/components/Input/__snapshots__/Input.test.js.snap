// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div>
  <form
    className="Message_border Left_right"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": "20px",
        }
      }
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
         
      </span>
    </button>
  </form>
</div>
`;

exports[`TextField component Matches the snapshot 1`] = `
<div>
  <form
    className="Message_border Left_right"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": "20px",
        }
      }
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
         
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
